io/zenoh-transport/src/unicast/establishment/mod.rs

```diff
@@ -116,3 +119,33 @@ pub(super) fn compute_sn(
     hasher.finalize_xof().read(&mut array);
     TransportSn::from_le_bytes(array) & seq_num::get_mask(resolution.get(Field::FrameSN))
 }
+
+pub(super) fn link_reliability(link: &LinkUnicast) -> Reliability {
+    if link.is_reliable() {
+        Reliability::Reliable
+    } else {
+        Reliability::BestEffort
+    }
+}
+
+pub(super) fn link_priority(
+    link: &LinkUnicast,
+    direction: TransportLinkUnicastDirection,
+) -> ZResult<Option<Priority>> {
+    const PRIORITY_METADATA_KEY: &str = "priority";
+
+    let link_locator = match direction {
```

Comment:
I believe setting the right priority would need both `src` and `dst` and find the min matching set. Moreover, I would assume that an extension to `Init/Open` should be added to communicate to the remote peer the actual `QoS` parameters should be associated with the link.

Reply:
What do you mean by "the min matching set"? The minimum of the `src` and `dst` priorities? If so then can't both peers agree on the priority tag of a link without exchanging further data?

Reply:
To agree on the priority an extension to `Init/Open` should be added.

Reply:
Priority ranges are now exchanged in `INIT` messages. Right now priorities are only applied if both sides agree on the exact range. I don't see a clear way unifying the two so I leave it for discussion:

https://github.com/fuzzypixelz/zenoh/blob/90b068f11293328da3dfffe59fa37fbed432d0ec/io/zenoh-transport/src/unicast/establishment/ext/qos.rs#L127-L135

Reply:
Also, endpoints are now recovered from the the "new link" channel and not from the `.get_src()` and `.get_dst()` methods.